File: 08-26-2025/QuickSort.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std;


int partition(vector<int>&arr,int start,int end){
    int idx=start-1,pivot=arr[end];
    for(int i=start;i<end; i++){
        if(arr[i]<pivot){
            idx++;
            swap(arr[i],arr[idx]);
        }
    }
    idx++;
    swap(arr[idx],arr[end]);
    return idx;
}

void quickSort(vector<int>&arr,int start,int end){
    for(int element: arr){
        cout<<element<<" ";
    }
    cout<<endl;
    if(start<end){
        int pivot=partition(arr,start,end);
        quickSort(arr,start,pivot-1);
        quickSort(arr,pivot+1,end);
    }
}

int main(){
    cout<<"Hello abhishek";
    vector<int>arr={4,3,6,2,7,1,8,0};
    quickSort(arr,0,arr.size()-1);
    return 0;
}

File: 08-26-2025/QuickSort.java
public class QuickSort{
    public static int partition(int arr[],int start, int end){
        int idx=start-1;
        int pivot=arr[end];
        for(int j=start; j<end; j++){
            if(arr[j]<=pivot){
                idx++;
                int temp=arr[j];
                arr[j]=arr[idx];
                arr[idx]=temp;
            }
        }
        idx++;
        int temp=arr[end];
        arr[end]=arr[idx];
        arr[idx]=temp;
        return idx;
    }
    public static void quickSort(int arr[],int start,int end){
        if(start<end){
            int pivotIdx=partition(arr,start,end);
            quickSort(arr, start, pivotIdx-1);
            quickSort(arr, pivotIdx+1, end);
        }
    }
    public static void main(String args[]){
        int arr[]={3,6,2,8,1,0};
        quickSort(arr,0,arr.length-1);
        for(int i=0;i<arr.length;i++){
            System.out.print(arr[i]+" ");
        }
    }
}